backend/backend-application/src/main/java/eu/socialedge/hermes/backend/application/api/service/PublicationService.java

```diff
@@ -0,0 +1,136 @@
+        return list(size, page, sort, filtering);
+    }
+
+    @Override
+    public ResponseEntity<PublicationDTO> getPublication(String id) {
+        return get(id);
+    }
+
+    @Override
+    public ResponseEntity<Resource> getPublicationContent(String id) {
+        val publication = repository.findOne(id);
+        if (publication == null) {
+            return ResponseEntity.notFound().build();
+        }
+        val fileType = publication.getStation() != null && publication.getLine() != null ? PDF_EXTENSION : ZIP_EXTENSION;
+        val filename = encode(format(FILE_NAME_TEMPLATE, publication.getName(), fileType));
+        val headers = new HttpHeaders();
+        headers.setContentType(MediaType.parseMediaType(format(MEDIATYPE_HEADER, fileType)));
+        headers.setContentDispositionFormData(filename, filename);
+        return new ResponseEntity<>(new ByteArrayResource(publication.getFile()), headers, HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<PublicationDTO> publishSchedules(PublicationSpecificationDTO dto) {
+        if (dto.getLineId() != null && !lineRepository.exists(dto.getLineId())) {
+            return ResponseEntity.notFound().build();
+        }
+        if (dto.getStationId() != null && !stationRepository.exists(dto.getStationId())) {
+            return ResponseEntity.notFound().build();
+        }
+        val line = Optional.ofNullable(dto.getLineId()).map(lineRepository::findOne).orElse(null);
+        val station = Optional.ofNullable(dto.getStationId()).map(stationRepository::findOne).orElse(null);
+        val schedules = scheduleRepository.findAll(dto.getScheduleIds());
+        byte[] fileContent;
+        String filename;
+        if (line != null && station != null) {
+            val document = timetableGenerationService.generateSingleLineStationTimetable(line, station, schedules);
+            fileContent = document.content();
+            filename = document.name();
+        } else if (line != null) {
+            val documents = timetableGenerationService.generateLineTimetables(schedules, line);
+            fileContent = Folder.of(documents).toZip();
+            filename = line.getName();
+        } else if (station != null) {
+            val documents = timetableGenerationService.generateStationTimetables(station, schedules);
+            fileContent = Folder.of(documents).toZip();
+            filename = station.getName();
+        } else {
+            return ResponseEntity.badRequest().build();
+        }
+        val schedulesList = StreamSupport.stream(schedules.spliterator(), false).collect(toList());
+        val publication = new Publication(filename, fileContent, schedulesList, line, station);
+        val persistedPublication = repository.save(publication);
+        return new ResponseEntity<>(mapper.toDTO(persistedPublication), HttpStatus.OK);
+    }
+
+    private static String encode(String filename) {
```

Comment:
I feel this function was used somewhere else, timetable gen maybe? If so, don't copy-paste :)

Reply:
It was used to encode file name returned by server. Timetable service (api) was deleted and endpoint for file retrieval is not in this class, where this method is used